crates/prices/src/core/estimate.rs

```diff
@@ -9,79 +10,86 @@ use {
 pub struct Estimate {
     /// The amount of [`ToToken`] that the user receives.
     pub amount: swap::ToAmount,
-    /// The amount of [`ToToken`] paid in fees.
+    /// The amount of [`ToToken`] paid in gas fees.
     pub fee: swap::ToAmount,
 }
 
+// TODO Revisit this doc comment
+/// An external token price estimator. Given a [`Swap`] (which specifies how
+/// much of the [`crate::FromToken`] should be converted into the
+/// [`crate::ToToken`]), the estimator returns an [`Estimate`] of how much of
+/// the [`crate::ToToken`] will be received, and how much will be paid in gas
+/// fees.
+pub trait Estimator: std::fmt::Debug + Send + Sync + 'static {
+    fn estimate(&self, swap: Swap) -> BoxFuture<'_, Result<(Price, eth::Gas), EstimatorError>>;
+}
+
 /// Estimate the outcome of a [`Swap`].
 pub async fn estimate(
     swap: Swap,
-    deadline: Deadline,
+    gas_price: eth::GasPrice,
     estimators: &[Box<dyn Estimator>],
 ) -> Result<Estimate, Error> {
-    // Fetch the estimate for the swap.
-    let estimate = median_estimate(swap, deadline, estimators).await?;
-
-    // Convert the fee from ETH to [`ToToken`].
-    let fee = median_estimate(
-        Swap {
-            from: eth::ETH_TOKEN.into(),
-            to: swap.to,
-            amount: estimate.gas.0.into(),
-        },
-        deadline,
-        estimators,
+    let ((swap_price, gas), (eth_price, _)) = try_join(
+        // Fetch the estimate for the swap price.
+        median_estimate(swap, estimators),
+        // Fetch the estimate for ETH price, to convert the gas fees into [`ToToken`].
+        median_estimate(
+            Swap {
+                // TODO Automatically wrap ETH, and specify the chain ID in the config
+                // from: eth::ETH_TOKEN.into(),
+                from: eth::H160::from_str("0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2")
+                    .unwrap()
+                    .into(),
+                to: swap.to,
+                amount: eth::U256::from(1000000000000000u64).into(),
```

Comment:
This amount needs to be configurable to account for the difference in price between eth on mainnet and xdai on gc.